### PR TITLE
DOKS: use GetCredentials for exec-credential

### PR DESCRIPTION
**Pull request by snormore**
Do less in the CLI and more in the API. Also properly support tokens.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -10,6 +10,8 @@ github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/digitalocean/godo v1.21.0 h1:oSgoZGague6FDPSRjhCiYd8s66QjkWJn+Jt4R/1kEyo=
 github.com/digitalocean/godo v1.21.0/go.mod h1:AAPQ+tiM4st79QHlEBTg8LM7JQNre4SAQCbn56wEyKY=
```
Comment (zachgersh) [inline go.sum]: think we need a mod tidy here to get rid of the old version?

File: commands/kubernetes.go [hunk 2]
```diff
@@ -759,22 +757,27 @@ func (s *KubernetesCommandService) RunKubernetesKubeconfigExecCredential(c *CmdC
...
+		ExpirationTimestamp:   &metav1.Time{Time: credentials.ExpiresAt},
+		Token:                 credentials.Token,
 	}
 
-	execCredential, err = execCredentialFromConfig(config)
-	if err != nil {
-		return err
+	execCredential = &clientauthentication.ExecCredential{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ExecCredential",
```
Comment (zachgersh) [inline commands/kubernetes.go]: probably should be a const? Will this ever change?

**Comment by snormore** [comment]
Thanks for the 👀 @zachgersh — comments should be addressed, 🙏

**Comment by zachgersh** [comment]
@snormore - could you hold off on merging this for a bit (we'll actually merge it for you). Want to fix a bug we've got first.

**Comment by snormore** [comment]
@zachgersh np, will leave it for you to merge when you're ready :+1:

**Pull request merged**